metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -300,26 +302,197 @@ private String postTestFile(String accession, int version) throws Exception {
+        mockMvc.perform(delete(location + "/taxonomies/" + idStr))
+                .andExpect(status().is4xxClientError())
+                .andExpect(jsonPath("exception").value("java.lang.IllegalArgumentException"))
+                .andExpect(jsonPath("message").value(ErrorMessage.SAMPLE_WITHOUT_TAXONOMY));
+    }
+
+    @Test
+    public void postSampleAndUpdateInvalidTaxonomies() throws Exception {
+        List<String> taxonomyUrlList = new ArrayList<String>();
+        String taxonomyUrl1 = postTestTaxonomy(1, "Species1");
+        taxonomyUrlList.add(taxonomyUrl1);
+        String taxonomyUrl2 = postTestTaxonomy(2, "Species2");
+        taxonomyUrlList.add(taxonomyUrl2);
+
+        String location = postTestSample("EGAN0001", "testSample", taxonomyUrlList);
 
         mockMvc.perform(get(location))
                 .andExpect(status().isOk())
                 .andExpect(jsonPath("$.accessionVersionId.accession").value("EGAN0001"))
-                .andExpect(jsonPath("$.name").value("testSample"));
+                .andExpect(jsonPath("$.name").value("testSample"))
+                .andExpect(jsonPath("$..taxonomies.href").value(location + "/" + "taxonomies"));
+
+        mockMvc.perform(get(location+ "/" + "taxonomies"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$..taxonomies").isArray())
+                .andExpect(jsonPath("$..taxonomies.length()").value(2))
+                .andExpect(jsonPath("$..taxonomies[0]..taxonomy.href").value(taxonomyUrl1))
+                .andExpect(jsonPath("$..taxonomies[1]..taxonomy.href").value(taxonomyUrl2));
+
+        List<String> taxonomyUrlListInvalid = new ArrayList<String>();
+        mockMvc.perform(patch(location)
+                .content("{ " +
+                        "\"taxonomies\": " + testListJson.write(taxonomyUrlListInvalid).getJson() + "" +
+                        "}"))
+                .andExpect(status().is4xxClientError())
+                .andExpect(jsonPath("exception").value("java.lang.IllegalArgumentException"))
+                .andExpect(jsonPath("message").value(ErrorMessage.SAMPLE_WITHOUT_TAXONOMY));
+
+        taxonomyUrlListInvalid.add("http://localhost/taxonomies/9998");
+        taxonomyUrlListInvalid.add("http://localhost/taxonomies/9999");
+        mockMvc.perform(patch(location)
+                .content("{ " +
+                        "\"taxonomies\": " + testListJson.write(taxonomyUrlListInvalid).getJson() + "" +
+                        "}"))
+                .andExpect(status().is4xxClientError())
+                .andExpect(jsonPath("exception").value("java.lang.IllegalArgumentException"))
+                .andExpect(jsonPath("message").value(ErrorMessage.INVALID_TAXONOMY));
+
+        List<String> taxonomyUrlListMixed = new ArrayList<String>();
+        taxonomyUrlListMixed.add(taxonomyUrl1);
+        taxonomyUrlListMixed.add("http://localhost/taxonomies/9999");
```

Comment:
Is there a way to refer to these URLs without specifying the host?

Reply:
Let me know